codeforboston/maple
# Privacy Setting - Adjusting "Browse Testimony"

# Summary
Closes #1431 
On Browse Testimony:
- Move the "Author" filter to the top of the filters
- Remove any user who has designated themselves as "private" from the "Author" filter
    - This is done by setting the authorDisplayName as "private" and then using transformitems on the RefinementList to filter the private users out of the RefinementList.
- For any "private user", their name should not be hyperlinked/clickable
- All users should have their full name shown 

# Checklist


# Screenshots
Jeff as a public user:
<img width="1511" alt="Screen Shot 2024-03-12 at 5 24 33 PM" src="https://github.com/codeforboston/maple/assets/55262612/b55f8a0d-04f5-43a5-84a2-afea8ba19106">
Jeff as a private user:
<img width="1512" alt="Screen Shot 2024-03-12 at 5 25 47 PM" src="https://github.com/codeforboston/maple/assets/55262612/fa20f7bd-eabd-48d5-b9e8-c525cd806f71">

# Known issues

- Since we use transformitems to filter out "private" from the RefinementList, on the off chance that a user signs up with a name as "private" it'll always filter them out of the list, no matter what their privacy status is.

# Steps to test/reproduce

_For each feature or bug fix, create a step by step list for how a reviewer can test it out. E.g.:_

1. Go to Browse Testimony and check if private users have their names filtered from the Author List
2. Go to profile and change privacy status to see if the hyperlinks and names appear in the Author list and Testimony results.
